### PR TITLE
fix: Support defaultExports in import rewriting

### PR DESCRIPTION
**Pull request by dpurp**
The import rewriter is not aware of default exports, which results in all exports being imported as a named export.

Supports #329

### CODE REVIEW

File: src/main/java/com/google/javascript/gents/ModuleConversionPass.java [hunk 1]
```diff
@@ -264,29 +294,32 @@ void convertRequireToImportStatements(Node n, String fullLocalName, String requi
 
     if (module.providesObjectChildren.get(requiredNamespace).size() > 0) {
```
Comment (gmoothart) [inline src/main/java/com/google/javascript/gents/ModuleConversionPass.java]: should this be an else if? It doesn't look like it makes sense to hit both code paths.

File: src/main/java/com/google/javascript/gents/ModuleConversionPass.java [hunk 2]
```diff
@@ -203,12 +205,16 @@ public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
 
   /**
    * Converts a Closure goog.require call into a TypeScript import statement.
-   *
+   * <p>
```
Comment (dpurp) [inline src/main/java/com/google/javascript/gents/ModuleConversionPass.java]: It's been quite some time; this was added by google-java-format.

File: src/main/java/com/google/javascript/gents/ModuleConversionPass.java [hunk 3]
```diff
@@ -242,17 +250,39 @@ void convertRequireToImportStatements(Node n, String fullLocalName, String requi
...
-      // import {value as localName} from "./file"
-      Node importSpec = new Node(Token.IMPORT_SPEC, IR.name(moduleSuffix));
-      // import {a as b} only when a =/= b
-      if (!moduleSuffix.equals(localName)) {
-        importSpec.addChildToBack(IR.name(localName));
+      @Nullable String defaultExportSymbol = module.exportedNamespacesToSymbols.get(EXPORTS);
+      String importedSymbol = module.importedNamespacesToSymbols.get(requiredNamespace);
+      boolean isDefaultExport = importedSymbol.equals(defaultExportSymbol);
+
+      Optional<Node> defaultExport;
```
Comment (dpurp) [inline src/main/java/com/google/javascript/gents/ModuleConversionPass.java]: I actually see that as an anti-pattern, since static analyzers are unable to determine if it was set in each branch.

File: src/test/java/com/google/javascript/gents/multiTests/type_rewrite/import.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import E from 'goog:keep.E';
 
 import * as A from './imported_module';
-import {A as X} from './imported_module';
+import X from './imported_module';
```
Comment (dpurp) [inline src/test/java/com/google/javascript/gents/multiTests/type_rewrite/import.ts]: We don't support the combined syntax, but I suspect that this still works

(e.g. `import defaultExport, {namedExports} from './file'`)

File: src/main/java/com/google/javascript/gents/ModuleConversionPass.java (hunk 1 above)
Comment (dpurp) [inline src/main/java/com/google/javascript/gents/ModuleConversionPass.java]: It'll break the existing test cases; though, those cases may not actually be valid.

In any case, since it changes existing functionality/support, it should be in its own PR.

File: src/main/java/com/google/javascript/gents/ModuleConversionPass.java (hunk 2 above)
Comment (mprobst) [inline src/main/java/com/google/javascript/gents/ModuleConversionPass.java]: By Google standards, it should be on the same line as the text, right?

```
/**
 * Foo bar baz.
 *
 * <p>Bam blergh.
 */
```

... not that I care.

**Comment by dpurp** [comment]
I agree with that philosophy for new TypeScript files, but lack of support for default exports makes it difficult to convert components one at a time, since (at least in our codebase) we tend to use default exports everywhere in Closure (`exports = Foo`).

Without support for default exports, we'll have to update the import/goog.require line for every client depending on our component, since they are outside of gents' scope.

**Comment by dpurp** [comment]
Withdrawing per discussions.
